### ISSUE TITLE
TransformConfigurationIndexIT#testDeleteConfigurationLeftOver fails reproducibly on 7.7

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#56761]
**Build scan**:

https://gradle-enterprise.elastic.co/s/w7ti6umvoycpm

**Repro line**:

`./gradlew ':x-pack:plugin:transform:qa:single-node-tests:integTestRunner' --tests "org.elasticsearch.xpack.transform.integration.TransformConfigurationIndexIT.testDeleteConfigurationLeftOver" -Dtests.seed=A895DD92B519091:54B8B45BCF6CAB0B -Dtests.security.manager=true -Dtests.locale=es-NI -Dtests.timezone=Asia/Anadyr -Dcompiler.java=14 -Druntime.java=8`

**Reproduces locally?**:

Yes, on 7.7 (209d9ad048eb738d8b8a4f19a887ed1983b62a59). No other failures recorded since https://github.com/elastic/elasticsearch/pull/55786 was merged 17 days ago.

**Applicable branches**:

Only observed on 7.7

**Failure excerpt**:

```
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:transform:qa:single-node-tests:integTestRunner' --tests "org.elasticsearch.xpack.transform.integration.TransformConfigurationIndexIT.testDeleteConfigurationLeftOver" -Dtests.seed=A895DD92B519091 -Dtests.security.manager=true -Dtests.locale=es-NI -Dtests.timezone=Asia/Anadyr -Dcompiler.java=14 -Druntime.java=8
  2> org.elasticsearch.client.ResponseException: method [GET], host [http://[::1]:63015], URI [.transform-notifications-000002/_search?ignore_unavailable=true], status line [HTTP/1.1 503 Service Unavailable]
    {"error":{"root_cause":[],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[]},"status":503}
        at __randomizedtesting.SeedInfo.seed([A895DD92B519091:54B8B45BCF6CAB0B]:0)
        at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:283)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:261)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:235)
        at org.elasticsearch.xpack.transform.integration.TransformRestTestCase.logAudits(TransformRestTestCase.java:506)
        at org.elasticsearch.xpack.transform.integration.TransformRestTestCase.waitForTransform(TransformRestTestCase.java:384)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
        at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
        at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by DaveCTurner** [comment]
Muted on 7.7 (only) with b2ec3db48db.

**Comment by hendrikmuhs** [comment]
@DaveCTurner 

This is https://github.com/elastic/elasticsearch/issues/54810, I just merged the 7.7 backport a few minutes ago (it was pending for a long time to not interfere with the release).

The build you reported on does not have the above backport.

**Comment by DaveCTurner** [comment]
Ah thanks for the quick response @hendrikmuhs I will revert the muting commit.
